mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,651 @@
+                y=self.y,
+                y_probs=self.y_probs,
+                labels=self.label_list,
+                curve_type="pr",
+            )
+
+            def plot_pr_curve():
+                pr_curve.plot_fn(**pr_curve.plot_fn_args)
+
+            self._log_image_artifact(plot_pr_curve, "precision_recall_curve_plot")
+            per_class_metrics_collection_df["precision_recall_auc"] = pr_curve.auc
+
+        self._log_pandas_df_artifact(per_class_metrics_collection_df, "per_class_metrics")
+
+    def _evaluate_classifier(self):
+        from mlflow.models.evaluation.lift_curve import plot_lift_curve
+
+        self.label_list = np.unique(self.y)
+        self.num_classes = len(self.label_list)
+
+        self.y_pred = self.predict_fn(self.X)
+        self.is_binomial = self.num_classes <= 2
+
+        if self.is_binomial:
+            if list(self.label_list) not in [[0, 1], [-1, 1]]:
+                raise ValueError(
+                    "Binary classifier evaluation dataset positive class label must be 1 or True, "
+                    "negative class label must be 0 or -1 or False, and dataset must contains "
+                    "both positive and negative examples."
+                )
+            _logger.info(
+                "The evaluation dataset is inferred as binary dataset, positive label is "
+                f"{self.label_list[1]}, negative label is {self.label_list[0]}."
+            )
+        else:
+            _logger.info(
+                "The evaluation dataset is inferred as multiclass dataset, number of classes "
+                f"is inferred as {self.num_classes}"
+            )
+
+        if self.predict_proba_fn is not None:
+            self.y_probs = self.predict_proba_fn(self.X)
+            if self.is_binomial:
+                self.y_prob = self.y_probs[:, 1]
+            else:
+                self.y_prob = None
+        else:
+            self.y_probs = None
+            self.y_prob = None
+
+        self.metrics.update(
+            _get_classifier_global_metrics(
+                self.is_binomial, self.y, self.y_pred, self.y_probs, self.label_list
+            )
+        )
+
+        if self.is_binomial:
+            self._log_binary_classifier()
+        else:
+            self._log_multiclass_classifier()
+
+        if self.is_binomial and self.y_probs is not None:
+            self._log_image_artifact(
+                lambda: plot_lift_curve(self.y, self.y_probs),
+                "lift_curve_plot",
+            )
+
+        # normalize the confusion matrix, keep consistent with sklearn autologging.
+        confusion_matrix = sk_metrics.confusion_matrix(
+            self.y, self.y_pred, labels=self.label_list, normalize="true"
+        )
+
+        def plot_confusion_matrix():
+            sk_metrics.ConfusionMatrixDisplay(
+                confusion_matrix=confusion_matrix,
+                display_labels=self.label_list,
+            ).plot(cmap="Blues")
```

Comment:
We need to scale these plots for class count > 6. The example for multiclass that has 10 classes is illegible with numeric wrapping overwrites of values. Setting figsize overrides on the canvas and changing the dpi at figure save time will make a much more clear visualization.

Reply:
The autologging code also generate confusion matrix plot for training dataset and has similar issue. Let's do a follow-up PR to improve for both cases.

Reply:
Create a follow-up ticket for this:
https://databricks.atlassian.net/browse/ML-19817